pkg/reconciliation/construct_statefulset.go

```diff
@@ -179,7 +179,7 @@ func newStatefulSetForCassandraDatacenter(
 	}
 	result.Annotations = map[string]string{}
 
-	if sts != nil && sts.Spec.ServiceName != result.Spec.ServiceName {
+	if sts != nil && sts.Spec.ServiceName != "" && sts.Spec.ServiceName != result.Spec.ServiceName {
```

Comment:
This would indicate that the dc.GetAllPodsServiceName() is returning empty string (set in the line 174)

Reply:
Wonder what would've created the empty sts originally

Reply:
Turns out we always start with an empty sts initially. Due to this bug for new CassandraDatacenters the `ServiceName` will always be empty string.

`dc.GetAllPodsServiceName()` does not return empty string.